ivy/func_wrapper.py

```diff
@@ -91,91 +91,48 @@ def caster(dtype, intersect):
                 return ret_dtype
 
 
+def cast_helper(arg, dtype, intersect, is_upcast=True):
+    step = 1 if is_upcast else -1
+    index = casting_modes_dict[arg]().index(dtype) + step
+    result = ""
+    if is_upcast:
+        while index < len(casting_modes_dict[arg]()):
```

Comment:
ig we could replace this with a single loop instead of 2 where the condition would change based on the value of `is_upcast`. What do you think?

Reply:
I thought of the same but I wasn't sure if `index` will be always >= 0. If the index value is always >= 0, then we can do the following:
```Python
result = ""
while 0 <= index < len(casting_modes_dict[t]()):
   if casting_modes_dict[t]()[index] not in intersect:
       result = casting_modes_dict[t]()[index]
       break
   index += index_step
return result
```
Please tell me your view, Thank you.

Reply:
I think it's safe to assume that `index` will always be `>=0` 😄